account.h

```diff
@@ -0,0 +1,39 @@
+#ifndef ACCOUNT_H_
+#define ACCOUNT_H_
+
+#include "person.h"
+#include "currency.h"
+#include "card_number.h"
+#include "pin.h"
+
+class Account
+{
+public:
+	Account(const CardNumber& card, const Pin& pin, 
+			const Currency& balance, const Person& owner);
+	~Account();
+
+	const	CardNumber& card() const;
+			CardNumber& card();
+	const	Pin& pin() const;
+			Pin& pin();
+	const	Currency& balance() const;
+			Currency& balance();
+	const	Person& owner() const;
+			Person& owner();
+
+	void withdraw(const Currency& amount);
+	void deposit(const Currency& amount);
+private:
+	CardNumber _card;
+	Pin _pin;
+	Currency _balance;
+	Person _owner;
+	bool _occupied;
```

Comment:
Can you explain, why does account need this field?

Reply:
It is possible that a few clients will try to deposit or withdraw money from the account simultaneously, and results can be incorrect. So before doing either of these operations, I will first check whether the account is `!occupied`. If `occupied`, then I'll wait, if not, I'll set it to `occupied` and perform the operation.